packages/components/src/theme/useMonacoTheme.tsx

```diff
@@ -0,0 +1,149 @@
+// react functional component
+
+import React, { useState, useRef, useEffect, useCallback } from 'react';
+import * as monaco from 'monaco-editor';
+import { ThemeContextType } from './ThemeProvider';
+
+/**
+ * Function to walk an object of key values, nested objects, and objects
+ * nested in arrays, and then update the values in a new object.
+ * @example
+ * const obj = {
+ *  rules: [{
+ *   token: 'keyword',
+ *   foreground: '--spectrum-blue-visual-color',
+ *   }],
+ * colors: {
+ *  'editor.background': '--spectrum-gray-50'
+ *  };
+ * const newColors = walkObject(obj, removeHashtag);
+ * console.log(newColors);
+ */
+function walkObject(
+  obj: { [key: string]: unknown },
+  fn: (v: unknown) => unknown
+) {
+  if (obj == null) return obj;
+  const newObj = obj;
+  Object.entries(newObj).forEach(([k, v]) => {
+    if (typeof v === 'object' && v !== null) {
+      walkObject(v as Record<string, unknown>, fn);
+    } else {
+      newObj[k] = fn(v);
+    }
+  });
+  return newObj;
+}
+
+/**
+ * Function to replace an unknown value with its computed value
+ * if it's a string and starts with '--'
+ * @param el The element to use to get the computed value
+ * @param variable The variable to replace
+ * @returns {any} The computed value or the original variable
+ * @example
+ * const el = document.querySelector('.spectrum');
+ * const computedValue = replaceCSSVariable(el, '--spectrum-alias-text-color');
+ * console.log(computedValue);
+ * // #ffffff
+ * @example
+ * const el = document.querySelector('.spectrum');
+ * const computedValue = replaceCSSVariable(el, '2');
+ * console.log(computedValue);
+ * // 2
+ */
+function replaceCSSVariable(el: Element, variable: unknown) {
```

Comment:
Is this actually replacing something or just deriving the value of a CSS variable?